datalad/interface/diff.py

```diff
@@ -65,12 +66,61 @@ def _translate_type(mode, ap, prop):
     elif mode == stat.S_IFDIR | stat.S_IFLNK:
         ap[prop] = 'dataset'
     elif stat.S_ISDIR(mode):
-        # not sure if this can happen at all
+        # not sure if this can happen at all, at least not in the tests...
         ap[prop] = 'directory'
     else:
         ap[prop] = 'file'
 
 
+def _get_untracked_content(dspath, report_untracked, paths=None):
+    cmd = ['git', '--work-tree=.', 'status', '--porcelain',
```

Comment:
I guess this call should be made available via *Repo.status(). Objections? Otherwise I'd make it an issue.